UnityProject/Assets/Scripts/NPC/AI/MobExplore.cs

```diff
@@ -38,10 +45,19 @@ public enum Target
 
 		public Target target;
 
+		public Vector3Int targetPos;
+
+		// to actually check if target is valid since vector isnt nullable type and we dont want for mobs to follow 0,0 or target far out of its target range
+		public bool targetIsValid = false;
```

Comment:
```suggestion
                [NonSerialized]
		public bool targetIsValid = false;
```

same with the other field above. You may also use a property and have the logic that handles if the taget is valid right there.

Reply:
nah, I dont think it actually worth moving logic here